packages/core/src/system.ts

```diff
@@ -1,22 +1,109 @@
 import {
   AnyEventObject,
-  ActorSystem,
   ActorSystemInfo,
   AnyActorRef,
   Observer,
-  Snapshot
+  Snapshot,
+  HomomorphicOmit,
+  EventObject
 } from './types.ts';
 
+export interface ScheduledEvent {
+  id: string;
+  event: EventObject;
+  startedAt: number; // timestamp
+  delay: number;
+  source: AnyActorRef;
+  target: AnyActorRef;
+}
+
+export interface Clock {
+  setTimeout(fn: (...args: any[]) => void, timeout: number): any;
+  clearTimeout(id: any): void;
+}
+
+export interface Scheduler {
+  schedule(
+    source: AnyActorRef,
+    data: {
+      id: string | undefined;
+      event: EventObject;
+      delay: number;
+      to?: AnyActorRef;
+    }
+  ): void;
+  cancel(id: string): void;
+  cancelAll(actorRef: AnyActorRef): void;
+}
+
+export interface ActorSystem<T extends ActorSystemInfo> {
+  /**
+   * @internal
+   */
+  _bookId: () => string;
+  /**
+   * @internal
+   */
+  _register: (sessionId: string, actorRef: AnyActorRef) => string;
+  /**
+   * @internal
+   */
+  _unregister: (actorRef: AnyActorRef) => void;
+  /**
+   * @internal
+   */
+  _set: <K extends keyof T['actors']>(key: K, actorRef: T['actors'][K]) => void;
+  get: <K extends keyof T['actors']>(key: K) => T['actors'][K] | undefined;
+  inspect: (observer: Observer<InspectionEvent>) => void;
+  /**
+   * @internal
+   */
+  _sendInspectionEvent: (
+    event: HomomorphicOmit<InspectionEvent, 'rootId'>
+  ) => void;
+  /**
+   * @internal
+   */
+  _relay: (
+    source: AnyActorRef | undefined,
+    target: AnyActorRef,
+    event: AnyEventObject
+  ) => void;
+  scheduler: Scheduler;
+  getSnapshot: () => {
+    _scheduledEvents: Record<string, ScheduledEvent>;
+  };
```

Comment:
we'll likely need `getPersistedSnapshot` here too - but that can be added when we manage to implement serialization/deserialization of actor refs